src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -316,22 +368,18 @@ def _monitor_workers_and_process_tasks_in_thread(
                 if crashed:
                     bulk_logger.warning(f"Process crashed while executing line {line_number}.")
                     ex = ProcessCrashError(line_number)
-                else:
+                elif self._line_timeout_expired(start_time, line_timeout_sec=line_timeout_sec):
                     # Handle line execution timeout.
```

Comment:
We may extract these part to something like `handle_output`

Reply:
Will refine in teh next PR.